src/main/java/org/jabref/model/search/rules/FullTextSearchRule.java

```diff
@@ -55,7 +57,9 @@ public PdfSearchResults getFulltextResults(String query, BibEntry bibEntry) {
             LOGGER.trace("Performing full query {}.", query);
             PdfIndexer pdfIndexer;
             try {
-                pdfIndexer = PdfIndexerManager.getIndexer(Globals.stateManager.getActiveDatabase().get(), Globals.prefs.getFilePreferences());
+                StateManager stateManager = Injector.instantiateModelOrService(StateManager.class);
+                PreferencesService preferencesService = Injector.instantiateModelOrService(PreferencesService.class);
+                pdfIndexer = PdfIndexerManager.getIndexer(stateManager.getActiveDatabase().get(), preferencesService.getFilePreferences());
```

Comment:
Use `Injector` inside `PdfIKndexerManger` to get an instance of `FilePrefererences` instead of using the constructor?

Reply:
Using contructor injection is always the cleanest solution for testing. About injection of smaller objects see above. Would require to abandon lazy loading of the preferences objects.

Reply:
I asked at jvm-german for the first point (https://jvm-german.slack.com/archives/C4HLX61PY/p1717757611057629).

For the second point, I don't fully understand as I thought that DI really leads to lazy loading, because the injection happens at the latest possible time (if the Injector is called directly). Let's first solve the general discussion injection via constructor vs. injection at variables vs. injection using direct calls. And then, the other things will also get solves surely.